crates/vm/levm/src/vm.rs

```diff
@@ -394,19 +407,23 @@ impl VM {
                 Err(error) => {
                     self.call_frames.push(current_call_frame.clone());
 
-                    // CONSUME ALL GAS UNLESS THE ERROR IS FROM REVERT OPCODE
+                    // Unless error is from Revert opcode, all gas is consumed
                     if error != VMError::RevertOpcode {
                         let left_gas = current_call_frame.gas_limit - current_call_frame.gas_used;
                         current_call_frame.gas_used += left_gas;
                         self.env.consumed_gas += left_gas;
                     }
 
+                    // Restore previous state
+                    (self.accrued_substate, self.db, self.env.refunded_gas) =
+                        (backup_substate, backup_db, backup_refunded_gas);
```

Comment:
Let's abstract this logic into a function

Reply:
Done!